avatars/avatars.js

```diff
@@ -1989,40 +2009,68 @@ class Avatar {
     // setup
     if (enabled) {
       this.volume = 0;
-     
+
       const audioContext = getAudioContext();
       if (audioContext.state === 'suspended') {
         (async () => {
           await audioContext.resume();
         })();
       }
-      this.microphoneWorker = new MicrophoneWorker({
-        audioContext,
-        muted: false,
-        emitVolume: true,
-        emitBuffer: true,
-      });
-      this.microphoneWorker.addEventListener('volume', e => {
-        if(!this.manuallySetMouth){
-          this.volume = this.volume*0.8 + e.data*0.2;
+      const _volume = e => {
+        if (!this.manuallySetMouth) {
+          this.volume = this.volume * 0.8 + e.data * 0.2;
+        } else {
+          console.log("couldn't set mouth because it is being manually set")
         }
-      });
-      this.microphoneWorker.addEventListener('buffer', e => {
+      }
+
+      const _buffer = e => {
         this.audioRecognizer.send(e.data);
-      });
+      }
 
       this.audioRecognizer = new AudioRecognizer({
         sampleRate: audioContext.sampleRate,
       });
+
       this.audioRecognizer.addEventListener('result', e => {
         this.vowels.set(e.data);
       });
+
+      this.audioWorker = new MicrophoneWorker({
+        audioContext,
+        muted: false,
+        emitVolume: true,
+        emitBuffer: true,
+      });
+
+      this.audioWorker.addEventListener('volume', _volume);
+      this.audioWorker.addEventListener('buffer', _buffer);
+
+      if (this.isLocalPlayer) {
+        this.microphoneWorker = new MicrophoneWorker({
+          audioContext,
+          muted: true,
+          emitVolume: true,
+          emitBuffer: true,
+        });
+
+        const _localVolume = e => {
+            this.volume = this.volume * 0.8 + e.data * 0.2;
+        }
+  
+        const _localBuffer = e => {
+          this.audioRecognizer.send(e.data);
+        }
+
+        this.microphoneWorker.addEventListener('volume', _localVolume);
+        this.microphoneWorker.addEventListener('buffer', _localBuffer);
+      }
     } else {
       this.volume = -1;
     }
   }
-  getAudioInput() {
-    return this.microphoneWorker && this.microphoneWorker.getInput();
+  getAudioInput(useMicrophone = false) {
```

Comment:
I don't think this should have an argument. If there are different nodes/different ways of inputting data to the avatar then there should be another method for grabbing an alternative node, rather than an argument switch.

Reply:
But I still don't get why there are two in the first place.

Reply:
Sounds like this should be `getLocalAudioInput` + `getRemoteAudioInput`.

The former would be for the microphone. The latter would also be for the case of the local hups talking.